server/src/main/java/org/opensearch/index/compositeindex/datacube/MetricStat.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+/**
+ * Supported metric types for composite index
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public enum MetricStat {
+    COUNT("count"),
```

Comment:
maybe we can have the type of the stat as a part of this enum. For eg: SUM will be of Double type